queue/server.go

```diff
@@ -0,0 +1,138 @@
+package queue
+
+import (
+	"sync"
+
+	"context"
+	"errors"
+	"time"
+
+	"github.com/centrifuge/gocelery"
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	BlockHeightParam string = "BlockHeight"
+	TimeoutParam     string = "Timeout"
+)
+
+var log = logging.Logger("queue-server")
+
+// Config is an interface for queue specific configurations
+type Config interface {
+
+	// GetNumWorkers gets the number of background workers to initiate
+	GetNumWorkers() int
+
+	// GetWorkerWaitTime gets the worker wait time for a task to be available while polling
+	// increasing this may slow down task execution while reducing it may consume a lot of CPU cycles
+	GetWorkerWaitTimeMS() int
+}
+
+// QueuedTaskType is a task to be queued in the centrifuge node to be completed asynchronously
+type QueuedTaskType interface {
```

Comment:
`QueuedTaskType` is not very intutive for me because Interface says it is type but the methods only has `TaskNameType`. I would go with `fmt.Stringer` for this

Reply:
Stringer is too generic to give any idea that it's an important part of the functionality of the queue. I want to make it explicit that this function is required to be implemented for task.